File: android/app/src/main/java/com/strangerevents/StrangerEventsService.java
package com.strangerevents;

import androidx.annotation.Nullable;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.modules.core.DeviceEventManagerModule;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.bridge.Arguments;

import java.util.Random;

public class StrangerEventsService extends ReactContextBaseJavaModule {

    StrangerEventsService(ReactApplicationContext context) {
        super(context);
    }

    @Override
    public String getName() {
        return "StrangerEventsService";
    }

    private void sendEvent(ReactContext reactContext,
                           String eventName,
                           @Nullable WritableMap params) {
        reactContext
                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                .emit(eventName, params);
    }


    @ReactMethod
    public void start() {
        Random random = new Random();
        int randomNumber = random.nextInt(17);

        if (randomNumber == 0) return;
        switch (randomNumber) {
            case 1:
                this.emmitEvent("Max", "73", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Max.jpeg");
                return;
            case 2:
                this.emmitEvent("Billy", "150", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Billy.jpeg");
                return;
            case 3:
                this.emmitEvent("Will", "79", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Will.jpeg");
                return;
            case 4:
                this.emmitEvent("Eleven", "999", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Eleven.jpeg");
                return;
            case 5:
                this.emmitEvent("Billy", "230", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Billy.jpeg");
                return;
            case 6:
                this.emmitEvent("Nancy", "22", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Nancy.jpeg");
                return;
            case 7:
                this.emmitEvent("Mike", "45", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Mike.jpeg");
                return;
            case 8:
                this.emmitEvent("Robin", "7", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Robin.jpeg");
                return;
            case 9:
                this.emmitEvent("Dustin", "330", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Dustin.jpeg");
                return;
            case 10:
                this.emmitEvent("Jim", "50", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Jim.jpeg");
                return;
            case 11:
                this.emmitEvent("Jonathan", "200", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Jonathan.jpeg");
                return;
            case 12:
                this.emmitEvent("Joyce", "84", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Joyce.jpeg");
                return;
            case 13:
                this.emmitEvent("Dr Brenner", "77", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Dr Brenner.jpeg");
                return;
            case 14:
                this.emmitEvent("Murray", "230", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Murray.jpeg");
                return;
            case 15:
                this.emmitEvent("Eddie", "100", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Eddie.jpeg");
                return;
            case 16:
                this.emmitEvent("Vecna", "0", "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Vecna.png");
                return;
            default:
                this.emmitEvent("", "0", "");
        }


    }

    public void emmitEvent (String name, String punctuation, String url ) {
        WritableMap params = Arguments.createMap();
        params.putString("name", name);
        params.putString("punctuation", punctuation);
        params.putString("url", url);

        sendEvent(this.getReactApplicationContext(), "strangeEvent", params);
    }



}

File: ios/Services/StrangerEventsService.swift
//
//  StrangerEventsService.swift
//  StrangerEvents
//
//  Created by Arthur Pasqualon on 03/07/22.
//
import React

import Foundation
@objc(StrangerEventsService)
class StrangerEventsService: RCTEventEmitter {
    
  override func supportedEvents() -> [String]! {
    return ["strangeEvent"]
  }
  
  @objc(start)
  func start() {
    
    let randomNumber = Int.random(in: 1...16)
        
    switch randomNumber {
    case 1:
      emmitEvent(name: "Max",punctuation: 73, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Max.jpeg")
    case 2:
      emmitEvent(name: "Billy",punctuation: 150, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Billy.jpeg")
    case 3:
      emmitEvent(name: "Will",punctuation: 79, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Will.jpeg")
    case 4:
      emmitEvent(name: "Eleven",punctuation: 999, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Eleven.jpeg")
    case 5:
      emmitEvent(name: "Billy",punctuation: 230, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Billy.jpeg")
    case 6:
      emmitEvent(name: "Nancy",punctuation: 22, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Nancy.jpeg")
    case 7:
      emmitEvent(name: "Mike",punctuation: 45, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Mike.jpeg")
    case 8:
      emmitEvent(name: "Robin",punctuation: 7, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Robin.jpeg")
    case 9:
      emmitEvent(name: "Dustin",punctuation: 330, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Dustin.jpeg")
    case 10:
      emmitEvent(name: "Jim",punctuation: 50, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Jim.jpeg")
    case 11:
      emmitEvent(name: "Jonathan",punctuation: 200, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Jonathan.jpeg")
    case 12:
      emmitEvent(name: "Joyce",punctuation: 84, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Joyce.jpeg")
    case 13:
      emmitEvent(name: "Dr Brenner",punctuation: 77, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Dr Brenner.jpeg")
    case 14:
      emmitEvent(name: "Murray",punctuation: 230, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Murray.jpeg")
    case 15:
      emmitEvent(name: "Eddie",punctuation: 100, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Eddie.jpeg")
    case 16:
      emmitEvent(name: "Vecna",punctuation: 0, link: "https://mixed-arthurpasqualon.s3.us-east-1.amazonaws.com/Vecna.png")
    default:
      emmitEvent(name: "",punctuation: 0, link: "")
    }
    
    
    return
  }
  
  
  @objc func emmitEvent(name: String, punctuation: Int, link: String) {
    self.sendEvent(
      withName: "strangeEvent",
      body: [
        "name": name,
        "url": link,
        "punctuation": punctuation
      ])
  }
  
 }
